relay-event-normalization/src/normalize/span/description/sql/parser.rs

```diff
@@ -274,12 +274,32 @@ impl VisitorMut for NormalizeVisitor {
             }
             Expr::BinaryOp {
                 ref mut left,
-                op: BinaryOperator::Or | BinaryOperator::And,
+                op: op @ (BinaryOperator::Or | BinaryOperator::And),
                 ref right,
             } => {
-                let is_equal = left == right;
-                if is_equal {
+                if left == right {
+                    //     /\
+                    //    /  \
+                    //   /\   B
+                    //  /  \
+                    // A    A
                     *expr = take_expr(left);
+                } else {
+                    //     /\
+                    //    /  \
+                    //   /\   B
+                    //  /  \
+                    // A    B
```

Comment:
A few questions, is the shape of the tree guaranteed to look like this? Are the following possible?

where the right side is a binary op
```
  /\
 /  \
A   /\
   /  \
  A    B
```

or both sides are a binary op

```
    / \
   /   \
  /\   /\
 /  \ /  \
A   B B   A
```

Reply:
@Zylphrex Excellent question! The parser deterministically parses multiple ORs / ANDs into a left-leaning binary tree, that is `a OR b OR c` is parsed into

```rust
BinaryOp {
    left: BinaryOp {
        left: Identifier(
            Ident {
                value: "a",
            },
        ),
        op: Or,
        right: Identifier(
            Ident {
                value: "b",
            },
        ),
    },
    op: Or,
    right: Identifier(
        Ident {
            value: "c",
        },
    ),
},
```

Your first example is only possible with parentheses. The second example is possible when the inner and outer operand are different.

```
    / \
   /   \
  /\   /\
 /  \ /  \
A   B B   A
```

could theoretically be normalized into

```
    / \
   /   \
  /\   /\
 /  \ /  \
A   B A   B
```

and then further into

```
   /\
  /  \
 A    B
```


but that goes beyond finding simple forms of repetition and into full algebraic normalization, so I would not touch it unless you think it's worth it.

Reply:
Understood! I don't think it's necessary as these are edge cases that I don't expect to happen frequently but it's good to know that it's something we can solve if it does occur